packages/client/src/components/app/Modal.svelte

```diff
@@ -0,0 +1,162 @@
+<script>
+  import { fade, fly } from "svelte/transition"
+  import { getContext } from "svelte"
+  import { clickOutside, Icon } from "@budibase/bbui"
+  import Portal from "svelte-portal"
+
+  const component = getContext("component")
+  const { styleable, modalStore, builderStore, dndIsDragging } =
+    getContext("sdk")
+
+  export let onClose
+  export let ignoreClicksOutside
+  export let size
+
+  // Open modal automatically in builder
+  $: {
+    if ($builderStore.inBuilder) {
+      if (
+        $component.inSelectedPath &&
+        $modalStore.contentId !== $component.id
+      ) {
+        modalStore.actions.open($component.id)
+      } else if (
+        !$component.inSelectedPath &&
+        $modalStore.contentId === $component.id &&
+        !$dndIsDragging
+      ) {
+        modalStore.actions.close()
+      }
+    }
+  }
+
+  $: open = $modalStore.contentId === $component.id
+
+  const handleModalClose = async () => {
+    if (onClose) {
+      await onClose()
```

Comment:
That sort of feature will tie in to something I've already been working on, which is prompting users before navigating away to avoid losing work, but I don't think we should make any changes for that at the moment.